sql/moz-fx-data-shared-prod/monitoring/structured_missing_columns_v1/query.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+"""Generates the list of missing columns in structured datasets."""
+import datetime
+import json
+from argparse import ArgumentParser
+from collections import defaultdict
+
+from google.cloud import bigquery
+
+NAMESPACE_QUERY = """
+SELECT
+  schema_name
+FROM
+  `moz-fx-data-shared-prod`.INFORMATION_SCHEMA.SCHEMATA
+WHERE
+  schema_name LIKE "%_stable"
+  AND schema_name NOT LIKE "%telemetry%"
+ORDER BY
+  schema_name
+"""
+
+QUERY = """
+-- missing columns {date} {namespace};
+WITH extracted AS (
+  SELECT
+    DATE "{date}" AS submission_date,
+    "{namespace}" AS document_namespace,
+    `moz-fx-data-shared-prod`.udf.extract_document_type(_TABLE_SUFFIX) AS document_type,
+    `moz-fx-data-shared-prod`.udf.extract_document_version(_TABLE_SUFFIX) AS document_version,
+    additional_properties
+  FROM
+    `moz-fx-data-shared-prod.{namespace}_stable.*`
+  WHERE
+    -- only observe full days of data
+    DATE(submission_timestamp) = date "{date}"
+    -- https://cloud.google.com/bigquery/docs/querying-wildcard-tables#filtering_selected_tables_using_table_suffix
+    -- IT's also possible to exclude tables in this query e.g
+    -- AND _TABLE_SUFFIX NOT IN ('main_v4', 'saved_session_v4', 'first_shutdown_v4')
+),
+transformed AS (
+  SELECT
+    * EXCEPT (additional_properties),
+    COUNT(*) AS path_count
+  FROM
+    extracted,
+    UNNEST(
+      `moz-fx-data-shared-prod`.udf_js.json_extract_missing_cols(
+        additional_properties,
+        [],
+        -- TODO: manually curate known missing schema bits
+        []
+      )
+    ) AS path
+  GROUP BY
+    submission_date,
+    document_namespace,
+    document_type,
+    document_version,
+    path
+)
+SELECT
+  submission_date,
+  document_namespace,
+  document_type,
+  document_version,
+  path,
+  path_count
+FROM
+  transformed
+"""
+
+# handy query for table sizes, although some tables should have less additional
+# properties than others.
+"""
+select *, round(byte_size/pow(1024,3), 2) as gb_size
+from monitoring.stable_table_sizes_v1
+where submission_date = "2020-11-11"
+and dataset_id <> "telemetry_stable"
+order by gb_size desc
+"""
```

Comment:
Do we need to keep this?

Reply:
No, I can leave it in my notes elsewhere.